### PR TITLE
feat: :sparkles: initial implementation of JsonlCorpora and Datasets

### LINKED ISSUE #2605: ✨ Add Jsonl corpus support
To be removed, once it is done: Please add the appropriate label to this ticket, e.g. feature or enhancement.

**Is your feature/enhancement request related to a problem? Please describe.**
For annotation of my texts, I mostly use doccano. Doccano also offers Jsonl as export format of the annotated datasets. I find this format very handy because it is easy to read and easy to process. It would be very convenient for me if flair supported Jsonl corpora.

**Describe the solution you'd like**
I have already implemented a JsonlCorpus for NER and text classification and would be willing to provide a PR if there is broader interest in a JsonlCorpus.

**Additional context**
Let me know if you are interested.

### PR DESCRIPTION
**Pull request by AnotherStranger**
I created an initial implementation for JSONL datasets using doccanos JSONL format. 
I tried to replicate the behavior of the existing ColumnCorpus.
I'd love to get feedback on the implementation.

Closes #2605

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 repos:
--   repo: https://github.com/ambv/black
+  - repo: https://github.com/ambv/black
     rev: stable
     hooks:
     - id: black
       language_version: python3.6
```
Comment (tadejmagajna) [inline .pre-commit-config.yaml:6]: bad indentation. Note that these changes to this file are already proposed in #2651
```suggestion
      - id: black
        language_version: python3.6 
```

File: flair/datasets/sequence_labeling.py [hunk 2]
```diff
@@ -16,6 +17,238 @@
...
+            raise ValueError(
+                f"Could not create token span from char span.\n\
+                    Sen: {sentence}\nStart: {start}, End: {end}, Label: {label}\n\
+                        Ann: {annotated_part}\nRaw: {text}\nCo: {start_idx}, {end_idx}"
+            )
+
+        # Add IOB tags
+        prefix = "B"
+        for token in sentence[start_idx : end_idx + 1]:
+            token.add_label("ner", f"{prefix}-{label}")
```
Comment (tadejmagajna) [inline flair/datasets/sequence_labeling.py:230]: Why does this code assume that label_type is always `ner`? What is you trying to do PoS tagging or any other type of labelling?

**Review by tadejmagajna — COMMENTED** [review]
Looks great! Like like how comprehensive the tests are.

The only question I have is why some parts of the code seem like they're hardcoded for `ner` `label_type ` only - it would be nice to have support for any type of tagging.

File: flair/datasets/sequence_labeling.py (hunk 2 above)
Comment (AnotherStranger) [reply flair/datasets/sequence_labeling.py:230]: Oh, my bad!
I forgot to parameterize this. I will change it ASAP.

Comment (AnotherStranger) [reply flair/datasets/sequence_labeling.py]: I added label_type as a parameter with the default value ner.

**Comment by alanakbik** [comment]
@AnotherStranger thanks a lot for adding this!

**Pull request merged**